app/model/build.js

```diff
@@ -1,13 +1,73 @@
+// Imports for packages
 import mongoose from 'mongoose';
+import Promise from 'bluebird';
 
+// Local imports
+import app from '~/';
+
+// Mongoose schema for builds
 const BuildSchema = mongoose.Schema({
   arch:       String,
   target:     String,
   version:    String,
-  started:    Date,
+  started: {
+    type:     Date,
+    default:  Date.now(),
+  },
   finished:   Date,
-  status:     String,
+  status: {
+    type:     String,
+    default: 'QUEUED',
+  },
   log:        String,
 });
 
+// Saves application with updated build information
+// Returns promise of saved application
+BuildSchema.methods.nestSave = function() {
```

Comment:
I was looking for information on this for soooo long. Apparently mongoose doesn't have any type of statics for nested documents. It also can't call save or update on nested documents, which provided to be a real headache. This biggest problem is when working with promises in parallel because when you call save on the parent document you can get duplicated or out of date data. This is why there are some weird looking mongoose functions in the application file. Using update $push and $set won't touch other data, making them safe to use in parallel.